### PR TITLE
BS-55: sort attributes alphanumerically

### PR DESCRIPTION
**Pull request by RangerRick**
Customer issue for sorting the list of custom resource graphs alphabetically (accounting for numerical sort in substrings).

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/BS-55

### CODE REVIEW

**Comment by j-white** [comment]
What's `BS`? :)

File: opennms-dao/src/main/java/org/opennms/netmgt/dao/support/RrdResourceAttributeUtils.java [hunk 1]
```diff
@@ -255,4 +256,79 @@ protected static String getStringProperty(File directory, String key) {
             throw new DataAccessResourceFailureException(message, e);
         }
     }
+
+    /**
+     * Alphanumeric sort that handles substrings with numeric components
+     */
+    public static class AlphaNumericOnmsAttributeComparator implements Serializable, Comparator<OnmsAttribute> {
```
Comment (j-white) [inline opennms-dao/src/main/java/org/opennms/netmgt/dao/support/RrdResourceAttributeUtils.java:263]: This is screaming at me for unit test coverage.

Comment (RangerRick) [reply opennms-dao/src/main/java/org/opennms/netmgt/dao/support/RrdResourceAttributeUtils.java:263]: > What's `BS`? :)

Fiiiine, it didn't seem worth making a whole other issue for a pretty simple change, but for you, I'll do it.  ;)

Yeah, I can add a couple unit tests.

Comment (RangerRick) [reply opennms-dao/src/main/java/org/opennms/netmgt/dao/support/RrdResourceAttributeUtils.java:263]: Didn't bother changing the branch name, but changed the commits.

File: opennms-dao/src/main/java/org/opennms/netmgt/dao/support/RrdResourceAttributeUtils.java [hunk 2]
```diff
@@ -255,4 +256,79 @@ protected static String getStringProperty(File directory, String key) {
...
+        public int compare(final OnmsAttribute firstAtt, final OnmsAttribute secondAtt) {
+            // Sorting algorithm found here: http://blog.icodejava.com/261/how-to-sort-alpha-numeric-strings-in-java/
+            final String firstString  = firstAtt.getName();
+            final String secondString = secondAtt.getName();
+
+            if (secondString == null || firstString == null) {
+                return 0;
+            }
+
+            int lengthFirstStr = firstString.length();
```
Comment (christianpape) [inline opennms-dao/src/main/java/org/opennms/netmgt/dao/support/RrdResourceAttributeUtils.java:275]: lengthFirstStr and lengthSecondStr can be final

File: opennms-dao/src/main/java/org/opennms/netmgt/dao/support/RrdResourceAttributeUtils.java [hunk 3]
```diff
@@ -255,4 +256,79 @@ protected static String getStringProperty(File directory, String key) {
...
+                        ch2 = secondString.charAt(index2);
+                    } else {
+                        break;
+                    }
+                } while (Character.isDigit(ch2) == Character.isDigit(space2[0]));
+
+                final String str1 = new String(space1);
+                final String str2 = new String(space2);
+
+                int result;
```
Comment (christianpape) [inline opennms-dao/src/main/java/org/opennms/netmgt/dao/support/RrdResourceAttributeUtils.java:316]: result can also be final

**Review by christianpape — APPROVED** [review]
Looks good.

**Pull request merged**